src/components/Home.jsx

```diff
@@ -1,19 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
 import getToken from '../lib/tokens';
 import { useHistory } from 'react-router-dom';
+import db from '../lib/firebase';
+import firebase from 'firebase/app';
+
+import './Home.css';
 
 const Home = () => {
   let history = useHistory();
+  const [tokenName, setTokenName] = useState('');
+  const [notification, setNotification] = useState(null);
 
   const generateToken = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+
+    db.collection('tokens')
+      .doc('tokenDocument')
+      .update({
+        tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
```

Comment:
Good question! What we decided to do was implement the optional task for your and Claire's issue last week, to store the token in the DB immediately on list creation. It goes in it's own 'tokens' collection. The token will still continue to be added to each item in the 'items' collection as before, so all of the code you and Zachee have written this week will still work perfectly. 
But with this method, the user can share an empty list straight away, which we thought was the best option. ☺️

Reply:
Hmm, I wonder why do we need to save tokens twice in the DB? Could we just query token through the `items` collection? I'm thinking, this could save us from doing extra operations of creating a new array, saving tokens, and querying another array.